packages/url_launcher/url_launcher_ios/ios/Classes/URLLauncherPlugin.swift

```diff
@@ -0,0 +1,116 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+
+public final class URLLauncherPlugin: NSObject, FlutterPlugin, UrlLauncherApi {
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let plugin = URLLauncherPlugin(binaryMessenger: registrar.messenger())
+    registrar.publish(plugin)
+  }
+
+  private var currentSession: URLLaunchSession?
+  private let launcher: Launcher
+
+  init(launcher: Launcher = UIApplicationLauncher(), binaryMessenger: FlutterBinaryMessenger) {
+    self.launcher = launcher
+    super.init()
+    UrlLauncherApiSetup.setUp(binaryMessenger: binaryMessenger, api: self)
```

Comment:
Why did you move this into `init` (making the test file more complicated)?

Reply:
Yeah, it is not necessary. I moved that out